hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/audit/AWSRequestAnalyzer.java

```diff
@@ -216,6 +216,18 @@ private RequestInfo writing(final String verb,
         || request instanceof CompleteMultipartUploadRequest
         || request instanceof GetBucketLocationRequest;
   }
+  /**
+   * Predicate which returns true if the request is part of the
+   * multipart upload API -and which therefore must be rejected
+   * if multipart upload is disabled.
+   * @param request request
+   * @return true if the transfer manager creates them.
+   */
+  public static boolean isRequestMultipartIO(final Object request) {
+    return request instanceof CopyPartRequest
+        || request instanceof CompleteMultipartUploadRequest
+        || request instanceof InitiateMultipartUploadRequest;
+  }
```

Comment:
I think we're missing `UploadPartRequest` for any new parts being uploaded.

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/UploadPartRequest.html

Reply:
oh yes. fixed. note that the list/abort uploads is not blocked either; left that alone as they are only used in specific circumstances
* s3a committers (which won't work here)
* s3guard uploads commmand
* if you ask the s3afs to cancel uploads on startup. Which is something we should cut as if it is turned on its dangerous